smart-contracts/advanced/deterministic-deployment.md

```diff
@@ -0,0 +1,141 @@
+}
+```
+
+### **`CREATE3`**
+
+With `CREATE2`, the contract bytecode affects the deployment address. So even blank spaces and comment text can affect the address.
+
+CREATE3 is similar to CREATE2 but without including the contract initCode on the address derivation formula. It can be used to generate deterministic contract addresses that aren’t tied to a specific contract code.
+
+CREATE3 is a way to use CREATE and CREATE2 in combination such that bytecode no longer affects the deployment address. — CREATE3 is more expensive than CREATE or CREATE2 (Fixed extra cost of ~55k gas).
+
+Check out a reference implementation [here](https://github.com/0xsequence/create3).
+
+### **Usage**
+
+Other people may have already deployed the factory contract onto some of your desired blockchains to the expected address (if they didn't change the deployment transaction data), in which case you won't need to deploy it on those blockchains - you can then just use those already-deployed factory contracts to deploy whatever other contracts you want to deploy. So first check the expected address on a blockchain explorer to see if a factory contract already exists there.
+
+If there isn't one yet then you'll need to deploy the factory contract via a **reusable signed raw deployment transaction**. The factory contract will then have the same address as on other blockchains (as long as the transaction bytecode stays the same). See the steps below to deploy the factory.
+
+1. **Prepare the deployment transaction:**
+    - Write the smart contract code for the factory contract.
+    - Compile the contract to get the bytecode.
+    - Create a deployment transaction with the bytecode.
+
+2. **Sign the deployment transaction:**
+    - Use a private key to sign the deployment transaction.
+    - Ensure the private key is securely stored and not exposed.
+
+3. **Broadcast the signed transaction:**
+    - Send the signed transaction to the desired blockchain network.
+    - Wait for the transaction to be mined and confirmed.
+
+4. **Verify the deployment:**
+    - Check the blockchain explorer to verify that the factory contract has been deployed to the expected address.
+    - Ensure the contract code matches the expected bytecode.
+
+By following these steps, you can deploy the factory contract to multiple blockchains, ensuring it has the same address on each one. This allows for consistent and deterministic deployment of other contracts using the factory contract.
+
+## **Popular Tools**
+
+1. [**Deterministic Deployment Proxy by Arachnid**](https://github.com/Arachnid/deterministic-deployment-proxy/tree/master)
+    - Signing the Proxy deployment tx is hidden, only recover the public key from signed tx.
+    - Somehow, the signing key and address is associated with the deployment tx. So if we change anything related to that deployment tx, the signing key and address will be different → proxy address is different.
+2. **Safe singleton Factory**
```

Comment:
Not sure what this section is meant to convey with the bullet points below (imo they don't help the reader). Iirc safe singleton factory is on Filecoin - would mention that and we can link to its docs so devs can refer to the address and usage directly.

Reply:
for consistency, pls hyperlink the numbered headers for tooling options e.g. safe's is not hyperlinked while arachnid and createx are